### PR TITLE
cohere[patch]: update embeddings and rerank

### PR DESCRIPTION
**Pull request by Anirudh31415926535**
This PR makes the model parameter mandatory for CohereEmbeddings class and the CohereRerank class.
Along with that, adds a embed method for the CohereEmbeddings class, and some minor improvements and docs updates.

### CODE REVIEW

File: libs/langchain-cohere/src/embeddings.ts [hunk 1]
```diff
@@ -70,8 +75,16 @@ export class CohereEmbeddings
       token: apiKey,
     });
     this.model = fieldsWithDefaults?.model ?? this.model;
+
+    if (!this.model) {
```
Comment (jacoblee93) [inline libs/langchain-cohere/src/embeddings.ts:79]: Would prefer to just make this a required param and ship a new minor version release along with your tool calling stuff

File: libs/langchain-cohere/src/embeddings.ts [hunk 2]
```diff
@@ -70,8 +75,16 @@ export class CohereEmbeddings
...
+      throw new Error(
+        "Model not specified for CohereEmbeddings instance. Please provide a model name from the options here: https://docs.cohere.com/reference/embed"
+      );
+    }
+
     this.batchSize = fieldsWithDefaults?.batchSize ?? this.batchSize;
-    this.inputType = fieldsWithDefaults?.inputType;
+    this.embeddingTypes =
+      fieldsWithDefaults?.embeddingTypes ?? this.embeddingTypes;
```
Comment (jacoblee93) [inline libs/langchain-cohere/src/embeddings.ts:87]: Do we need to remove `inputType` from `CohereEmbeddingsParams` as well?

File: libs/langchain-cohere/src/embeddings.ts [hunk 3]
```diff
@@ -137,6 +154,25 @@ export class CohereEmbeddings
     }
   }
 
+  async embed(
```
Comment (jacoblee93) [inline libs/langchain-cohere/src/embeddings.ts:157]: Why add this? Would prefer not to extend base class' API unnecessarily.

**Comment by jacoblee93** [comment]
Thanks, see comments!

Let's gear up for a minor version release along with your other PR.

File: libs/langchain-cohere/src/embeddings.ts (hunk 3 above)
Comment (Anirudh31415926535) [reply libs/langchain-cohere/src/embeddings.ts:157]: This method is supported in langchain-python, and also allows to override the inputType parameter to call the embed function

**Pull request merged**